linera-execution/src/system.rs

```diff
@@ -389,14 +389,26 @@ pub enum SystemExecutionError {
     #[error("Chain is not active yet.")]
     InactiveChain,
 
-    #[error("Blob not found on storage read: {0}")]
-    BlobNotFoundOnRead(BlobId),
+    #[error("Blobs not found: {0:?}")]
+    BlobsNotFound(Vec<BlobId>),
     #[error("Oracle response mismatch")]
     OracleResponseMismatch,
     #[error("No recorded response for oracle query")]
     MissingOracleResponse,
 }
 
+impl SystemExecutionError {
+    pub fn get_blobs_not_found(&self) -> Option<Vec<BlobId>> {
```

Comment:
In the rest of the code, we try to normalize the errors during conversion instead

Reply:
(There's also https://github.com/linera-io/linera-protocol/issues/2628 for this.)

Reply:
Yes, I agree this is not the ideal end solution for this, this is just an intermediate step until we get to #2628

Reply:
Let's add a `TODO(#2628)` to both `get_blobs_not_found` functions.

Reply:
Talked with @afck a while back, I'm just gonna do #2628 here already